pagopa/io-app
# fix: [#175822523] Fix time format in MessageListItem

## Short description
The time format is now correct.

## List of changes proposed in this pull request
- Time hours format is changed according to specs
- Removed unnecessary else case, we don't need to handle invalid dates as they are validated by the BE
- Added proper unit tests

## How to test
Run the app, open the "Messages" tab, see the time formatted like "13:15" instead of "13.15"
